doc/source/user_guide/numpy_images.txt

```diff
@@ -55,28 +64,134 @@ Masking (indexing with masks of booleans)::
 disk: ::
 
-    >>> l_x, l_y = camera.shape[0], camera.shape[1]
-    >>> X, Y = np.ogrid[:l_x, :l_y]
-    >>> outer_disk_mask = (X - l_x / 2)**2 + (Y - l_y / 2)**2 < (l_x / 2)**2
+    >>> nrows, ncols = camera.shape
+    >>> row, col = np.ogrid[:nrows, :ncols]
+    >>> cnt_row, cnt_col = nrows / 2, ncols / 2
+    >>> outer_disk_mask = ((row - cnt_row)**2 + (col - cnt_col)**2 <
+    ...                    (nrows / 2)**2)
     >>> camera[outer_disk_mask] = 0
 
 .. image:: ../../_images/plot_camera_numpy_1.png
     :width: 45%
     :target: ../auto_examples/plot_camera_numpy.html
 
-Boolean arithmetics can be used to define more complex masks: ::
+Boolean arithmetic can be used to define more complex masks: ::
 
-    >>> lower_half = X > l_x / 2
+    >>> lower_half = row > cnt_row
     >>> lower_half_disk = np.logical_and(lower_half, outer_disk_mask)
     >>> camera = data.camera()
     >>> camera[lower_half_disk] = 0 
+
+
+Color images
+------------
+
+All of the above is true of color images, too: a color image is a
+NumPy array, with an additional trailing dimension for the channels:
+
+    >>> cat = data.chelsea()
+    >>> type(cat)
+    <type 'numpy.ndarray'>
+    >>> cat.shape
+    (300, 451, 3)
+
+That's a 300-by-451 pixel image with red, green, and blue channels.
+As before, we can get and set pixel values:
+
+    >>> cat[10, 20]
+    array([151, 129, 115], dtype=uint8)
+    >>> # set the pixel at row 50, column 60 to black
+    >>> cat[50, 60] = 0
+    >>> # set the pixel at row 50, column 61 to green
+    >>> cat[50, 61] = [0, 255, 0] # [red, green, blue]
+
+We can also use 2D boolean masks for a 2D color image, as we did with
+the grayscale image above:
+
+.. plot::
+
+    Using a 2D mask on a 2D color image
+
+    >>> reddish = cat[:, :, 0] > 160
+    >>> cat[reddish] = [0, 255, 0]
+    >>> plt.imshow(cat)
+
+
+Coordinate conventions
+----------------------
+
+Because we represent images with numpy arrays, our coordinates must
+match accordingly. Two-dimensional (2D) grayscale images (such as
+`camera` above) are indexed by row and columns (abbreviated to either
+``row, col`` or ``r, c``), with the lowest element (0, 0) at the top-
+-left corner. In various parts of the library, you will
+also see ``rr`` and ``cc`` refer to lists of row and column
+coordinates. We distinguish this from (x, y), which commonly denote
+Cartesian coordinates, where x is the horizontal coordinate, y the
+vertical, and the origin is on the bottom right. (Matplotlib, for
+example, uses this convention.)
+
+In the case of color (or multichannel) images, the last dimension
+contains the color information and is denoted ``channel`` or ``ch``.
+
+Finally, for 3D images, such as videos, magnetic resonance imaging
+(MRI) scans, or confocal microscopy, we refer to the leading dimension
+as ``frame``, abbreviated as ``frm`` or ``f``. In many cases,
+the third imaging dimension has lower resolution than the other two,
```

Comment:
so, for 3-D images such as MRI or tomography, you suggest to index as im[fm, rr, cc] ? This convention has the drawback to single out one dimension, which often is no more than a dimension of space.